pkg/volume/scaleio/sio_util.go

```diff
@@ -79,23 +85,44 @@ var (
 	protectionDomainNotProvidedErr = errors.New("ScaleIO protection domain not provided")
 )
 
-// mapScaleIOVolumeSource maps attributes from a ScaleIOVolumeSource to config
-func mapVolumeSource(config map[string]string, source *api.ScaleIOVolumeSource) {
-	config[confKey.gateway] = source.Gateway
-	config[confKey.secretRef] = func() string {
-		if source.SecretRef != nil {
-			return string(source.SecretRef.Name)
-		}
-		return ""
-	}()
-	config[confKey.system] = source.System
-	config[confKey.volumeName] = source.VolumeName
-	config[confKey.sslEnabled] = strconv.FormatBool(source.SSLEnabled)
-	config[confKey.protectionDomain] = source.ProtectionDomain
-	config[confKey.storagePool] = source.StoragePool
-	config[confKey.storageMode] = source.StorageMode
-	config[confKey.fsType] = source.FSType
-	config[confKey.readOnly] = strconv.FormatBool(source.ReadOnly)
+// mapVolumeSpec maps attributes from either ScaleIOVolumeSource  or ScaleIOPersistentVolumeSource to config
+func mapVolumeSpec(config map[string]string, spec *volume.Spec) {
+
+	if source, err := getScaleIOPersistentVolumeSourceFromSpec(spec); err == nil {
+		config[confKey.gateway] = source.Gateway
+		config[confKey.secretName] = func() string {
+			if source.SecretRef != nil {
+				return string(source.SecretRef.Name)
+			}
+			return ""
+		}()
```

Comment:
Not sure how this is used, but should set config[confKey.secretNamespace] from source.SecretRef as well, right?

Reply:
Good catch, secretName and namespace are set outside of this function anyway, removing it.